packages/repository/src/repositories/unsafe-crud-repository.ts

```diff
@@ -0,0 +1,186 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {
+  DataObject,
+  Options,
+  Command,
+  NamedParameters,
+  PositionalParameters,
+  AnyObject,
+} from '../common-types';
+import {Entity} from '../model';
+import {Filter, Where} from '../query';
+import {
+  CrudRepository,
+  EntityCrudRepository,
+  EntityRepository,
+} from './repository';
+import {isEntityNotFoundError} from '../errors';
+
+/**
+ * CRUD operations for a repository of entities.
+ *
+ * Operations affecting a single entity signal "not found" error by returning
+ * `undefined`, as opposed to EntityCrudRepository that throws EntityNotFound
+ * error.
+ */
+export interface UnsafeCrudRepository<T extends Entity, ID>
```

Comment:
We should name it something like `UncheckedEntityRepository`. `Unsafe` really scares me :-).

Reply:
Good suggestion, I agree `Unchecked` is better than `Unsafe`.

Here is my reasoning why I prefer to exclude `Entity` and include `Crud` in the interface name:
 - I feel that different requirements for handling "does not exist" are specific to entities, because they have an `id` that's used to lookup an existing record. I don't see how a "not found" situation can happen for a model that does not have any unique identifier. As a result, all "unchecked" repositories are going to deal with Entities.
 - I am assuming there is a reason why we have `EntityCrudRepository` instead of shorter `EntityRepository` and we are envisioning to have non-CRUD entity repositories in the future (or in 3rd party code). Maybe KeyValue is a good example? As a result, I want to make it clear that the interface I am implementing here is providing CRUD access.

Going forward, I'll use the name `UncheckedCrudRepository`.